### PR TITLE
various name resolving related improvements

### PR DESCRIPTION
**Pull request by Geertiebear**
Crosses off 4/5 points currently listed on issue #99, mainly:

- add ``/etc/services`` parsing
- add ``/etc/resolv.conf`` parsing
- add ``/etc/hosts`` parsing
- deal with CNAME dns cleaning up the alias related code

### CODE REVIEW

File: options/posix/generic/netdb-stubs.cpp [hunk 1]
```diff
@@ -53,52 +55,63 @@ int getaddrinfo(const char *__restrict node, const char *__restrict service,
...
-	if (ret < 0)
-		return ret;
-
-	auto out = (struct mlibc::ai_buf *) calloc(ret, sizeof(struct addrinfo));
-
-	auto canon_len = strlen(node);
-	auto canon = (char *) malloc(canon_len);
-	strncpy(canon, node, canon_len);
+	struct mlibc::service_buf serv_buf[SERV_MAX] = {};
+	int serv_count = mlibc::lookup_serv(serv_buf, service, protocol, socktype);
```
Comment (avdgrinten) [inline options/posix/generic/netdb-stubs.cpp:65]: It's guaranteed that this function only ever writes SERV_MAX = 2 entries? (If not, we could use a frg::small_vector or so.)

File: options/posix/generic/services.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+			break;
+		}
+		if (!pos)
+			continue;
+
+		// now that we know the name is in this line we can parse
+		// the rest
+
+		// skip the name at the beginning of the line
+		for(pos = line; *pos && !isspace(*pos); pos++);
```
Comment (avdgrinten) [inline options/posix/generic/services.cpp:54]: Put the `;` onto the next line and indent it.

File: options/posix/include/mlibc/lookup.hpp [hunk 3]
```diff
@@ -18,6 +18,13 @@ struct dns_addr_buf {
 	uint8_t addr[16];
 };
 
+struct lookup_result {
+	lookup_result() : buf(getAllocator()), aliases(getAllocator())
+	{}
```
Comment (avdgrinten) [inline options/posix/include/mlibc/lookup.hpp:23]: Coding style wise:
```suggestion
	lookup_result()
	: buf(getAllocator()), aliases(getAllocator()) {}
```
Also, `{}` initialization might be preferable to `()` initialization in general.

File: options/posix/generic/netdb-stubs.cpp (hunk 1 above)
Comment (Geertiebear) [reply options/posix/generic/netdb-stubs.cpp:65]: yeah it is

**Pull request merged**